### PR TITLE
add section why cache limitation is necessary with examples #70

### LINKED ISSUE #70: Service worker not updating without force update
Hi,

I installed Vue PWA template and tested on Apache localhost with offline feature (Dev tool). It works as expected.
I deployed same code to the production environment on [Heroku.](https://service-worker-poc-2.herokuapp.com/) where it is not working as expected.
Even when I reload the page old service worker is still running and cached version of app which I get in offline mode is stale.
When I force update the service worker (From dev tool) then newer version of service worker is installed and only then it fetches fresh copy of project.

Here's my production webpack config.

```
    new SWPrecacheWebpackPlugin({
      cacheId: 'my-vue-app',
      filename: 'service-worker.js',
      staticFileGlobs: ['dist/**/*.{js,html,css}'],
      minify: true,
      stripPrefix: 'dist/',
      mergeStaticsConfig: true, // if you don't set this to true, you won't see any webpack-emitted assets in your serviceworker config
      staticFileGlobsIgnorePatterns: [/\.map$/], // use this to ignore sourcemap files
    })
```
I have not changed service worker code in index.html

Is there anything I am missing?

### PR DESCRIPTION
**Pull request by ahus1**
This tries to document the necessary web server setup regarding caching. This should close #70

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,6 +22,12 @@ $ npm run dev
 
 If port 8080 is already in use on your machine you must change the port number in `/config/index.js`. Otherwise `npm run dev` will fail.
 
+## What's not Included
+
+* You'll need to configure your web server to add HTTP headers to prevent caching of critical service worker files.
+If you don't do this, you might end up with browsers that never update the cached content.
```
Comment (jeffposnick) [inline README.md]: It's not (quite) as bad as that.

Can you update the "never" to say that there might be a delay of up to 24 hours? And link to https://stackoverflow.com/questions/38843970/service-worker-javascript-update-frequency-every-24-hours/38854905#38854905

File: README.md [hunk 2]
```diff
@@ -63,3 +69,46 @@ You can fork this repo to create your own boilerplate, and use it with `vue-cli`
 ``` bash
 vue init username/repo my-project
 ```
+
+## Configuring your Web server to prevent caching
+
+When you run the production build and deploy the application to a web server, the browser will call the service worker on the first visit and will download and cache all resources. 
+When the browser accesses the site a second time, it will use the cached resources.
+To notice changes in your app it is crucial that the browser downloads the latest version `service-worker.js` from your server.
+The browser will only do this when the HTTP Header of the service worker indicates that a previous download has expired.
```
Comment (jeffposnick) [inline README.md]: Can you rephrase "HTTP Header" to "[Cache-Control](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/http-caching) header associated with `service-worker.js`"

File: README.md [hunk 3]
```diff
@@ -63,3 +69,46 @@ You can fork this repo to create your own boilerplate, and use it with `vue-cli`
...
+If a new service worker is available, it will download the new contents in the background.
+Once the download is complete, the default behavior is that the new content will be shown on the next visit of the site. 
+You can change the template's `service-worker-prod.js` to display the message to the user to reload the page, or issue a `window.location.reload()` automatically to trigger an update depending on your application's needs.
```
Comment (jeffposnick) [inline README.md]: `service-worker-prod.js` => [`service-worker-prod.js`](https://github.com/vuejs-templates/pwa/blob/master/template/build/service-worker-prod.js)

File: README.md [hunk 4]
```diff
@@ -63,3 +69,46 @@ You can fork this repo to create your own boilerplate, and use it with `vue-cli`
...
+
+Depending on your setup, you might want to have a short caching time for your service worker or no caching at all.
+At minimum, you should add caching headers for `service-worker.js` and `index.html`.
```
Comment (jeffposnick) [inline README.md]: Can you rephrase "you should add caching headers" to make it clear that the headers you add should disable caching?

File: README.md [hunk 5]
```diff
@@ -63,3 +69,46 @@ You can fork this repo to create your own boilerplate, and use it with `vue-cli`
...
+You should also add caching headers for all files that are in the `static` folder of this template.
```
Comment (jeffposnick) [inline README.md]: Link to https://jakearchibald.com/2016/caching-best-practices/

**Comment by ahus1** [comment]
@jeffposnick - Hello Jeff, thank you for the review. I've added the new content. I've also move the some of the content I added to README.md to a file docs/prevent_caching.md. 

Please review again.

**Comment by jeffposnick** [comment]
Thanks!
